DigitalLearningSolutions.Web.Tests/Controllers/LearningMenu/LearningMenuControllerTests.cs

```diff
@@ -574,6 +576,193 @@ public void Diagnostic_assessment_should_404_if_failed_to_enrol()
+            // Given
+            const int progressId = 299;
+            var selectedTutorials = new List<int>();
+            var defaultDiagnosticContent = DiagnosticAssessmentHelper.CreateDefaultDiagnosticContent();
+            A.CallTo(() => diagnosticAssessmentService.GetDiagnosticContent(CustomisationId, SectionId))
+                .Returns(defaultDiagnosticContent);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId)).
+                Returns(progressId);
+
+            // When
+            var result = controller.DiagnosticContent(CustomisationId, SectionId, EmptyTutorialSelection);
+
+            // Then
+            var expectedModel = new DiagnosticContentViewModel(config, defaultDiagnosticContent, selectedTutorials, CustomisationId, CentreId, SectionId, progressId, CandidateId);
+            result.Should().BeViewResult()
+                .Model.Should().BeEquivalentTo(expectedModel);
+        }
+
+        [Test]
+        public void Diagnostic_content_should_404_if_diagnostic_content_not_found()
+        {
+            // Given
+            A.CallTo(() => diagnosticAssessmentService.GetDiagnosticContent(CustomisationId, SectionId))
+                .Returns(null);
+
+            // When
+            var result = controller.DiagnosticContent(CustomisationId, SectionId, EmptyTutorialSelection);
+
+            // Then
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(A<int>._, A<int>._, A<int>._)).MustNotHaveHappened();
+            result.Should()
+                .BeRedirectToActionResult()
+                .WithControllerName("LearningSolutions")
+                .WithActionName("StatusCode")
+                .WithRouteValue("code", 404);
+        }
+
+        [Test]
+        public void Diagnostic_content_should_404_if_failed_to_enrol()
+        {
+            // Given
+            var defaultDiagnosticContent = DiagnosticAssessmentHelper.CreateDefaultDiagnosticContent();
+            A.CallTo(() => diagnosticAssessmentService.GetDiagnosticContent(CustomisationId, SectionId))
+                .Returns(defaultDiagnosticContent);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId)).Returns(null);
+
+            // When
+            var result = controller.DiagnosticContent(CustomisationId, SectionId, EmptyTutorialSelection);
+
+            // Then
+            result.Should()
+                .BeRedirectToActionResult()
+                .WithControllerName("LearningSolutions")
+                .WithActionName("StatusCode")
+                .WithRouteValue("code", 404);
+        }
+
+        [Test]
+        public void Diagnostic_content_should_404_if_tutorial_id_not_found()
+        {
+            // Given
+            const int progressId = 299;
+            var availableTutorials = new List<int>(new[] { 1, 2, 3 });
+            var selectedTutorials = new List<int>(new[] { 4, 5, 6 });

```

Comment:
I think we should also test the scenario where there's some overlap between the two. And do we have a test where the selected tutorials are a subset of the available tutorials (which should be allowed)? 🙂